pyiceberg/catalog/rest.py

```diff
@@ -447,13 +447,14 @@ def create_table(
         sort_order: SortOrder = UNSORTED_SORT_ORDER,
         properties: Properties = EMPTY_DICT,
     ) -> Table:
-        schema: Schema = self._convert_schema_if_needed(schema)  # type: ignore
+        iceberg_schema = self._convert_schema_if_needed(schema)
+        iceberg_schema = assign_fresh_schema_ids(iceberg_schema)
 
         namespace_and_table = self._split_identifier_for_path(identifier)
         request = CreateTableRequest(
             name=namespace_and_table["table"],
             location=location,
-            table_schema=schema,
+            table_schema=iceberg_schema,
```

Comment:
Isn't this done on the server side?

Reply:
Yes, but some server sides still require the IDs to be unique